### PR TITLE
Fix incorrect inputs into tf.random.categorical

### PR DESCRIPTION
**Pull request by HJasperson**
change first input to log probabilities rather than logits

## Why are these changes needed?

The inputs for tf.random.categorical are incorrect. The name of the input is 'logits', but it actually represents the log probabilities according to [the documentation](https://www.tensorflow.org/api_docs/python/tf/random/categorical#args).

With the current inputs, tf.random.categorical is free to sample from masked actions, which should have zero probability.

## Related issue number

#24055 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/models/tf/tf_action_dist.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def kl(self, other: ActionDistribution) -> TensorType:
 
     @override(TFActionDistribution)
     def _build_sample_op(self) -> TensorType:
-        return tf.squeeze(tf.random.categorical(self.inputs, 1), axis=1)
+        return tf.squeeze(tf.random.categorical(tf.math.log(tf.nn.softmax(tf.where(self.inputs!=0, self.inputs, tf.float64.min))), 1), axis=1)
```
Comment (gjoliver) [inline rllib/models/tf/tf_action_dist.py:91]: please use a local varialbe:
```
logp = tf.math.log(tf.nn.softmax(tf.where(self.inputs!=0, self.inputs, tf.float64.min)))
tf.squeeze(tf.random.categorical(logp, 1), axis=1)
```

Comment (gjoliver) [reply rllib/models/tf/tf_action_dist.py:91]: I also don't get why we need to modify the logits here, shouldn't we fix action_masking to use tf.float64.min instead?

**Review by gjoliver — COMMENTED** [review]
feel like we need to benchmark a bunch of things

File: rllib/models/tf/tf_action_dist.py (hunk 1 above)
Comment (HJasperson) [reply rllib/models/tf/tf_action_dist.py:91]: there's 2 reasons to modify the logits here:
1. leaving a lot of (what is essentially) -inf around can cause nan problems with algos like ppo
2. action_masking doesn't catch all cases (e.g. people will do it some other way or have a more complex situation)
